fennel/featuresets/test_invalid_featureset.py

```diff
@@ -377,11 +378,81 @@ class A:
 
         @featureset
         class B:
-            user_id: int = F(A.user_id, default=0)
+            user_id: int = feature(A.user_id, default=0)
             home_geoid: int
             credit_score: int
 
     assert (
         str(e.value)
         == "'Please specify a reference to a field of a dataset to use \"default\" param', found arg: `user_id` and default: `0`"
     )
+
+
+@mock
+def test_invalid_expr_feature(client):
+
+    # Using a feature that is not defined in the featureset
+    with pytest.raises(ValueError) as e:
+
+        @featureset
+        class UserInfo3:
+            user_id: int
+            home_geoid: int
+            age: int = feature(UserInfoDataset.age, default=0)
+            age_squared: int = feature(F("Age") * F("Age"))
+            credit_score: int
+
+    assert (
+        str(e.value)
+        == "When using F(<feature_name>) for expressions, one can only choose from the features defined in the current featureset. Please choose an input from : ['user_id', 'home_geoid', 'age', 'age_squared', 'credit_score'] found : `Age` in extractor : `_fennel_expr_UserInfo3.age_squared`."
+    )
+
+    # Using default value for an expression feature
+    with pytest.raises(ValueError) as e:
+
+        @featureset
+        class UserInfo4:
+            user_id: int
+            home_geoid: int
+            age: int = feature(UserInfoDataset.age, default=0)
+            age_squared: int = feature(F("age") * F("age"), default=0)
+            credit_score: int
+
+    assert (
+        str(e.value)
+        == "Default value cannot be set for an expression, use fillnull instead, found : Ref('age') * Ref('age')"
+    )
+
+    # Incorrect type for an expression feature
+    with pytest.raises(TypeError) as e:
+
+        @featureset
+        class UserInfo5:
+            user_id: int
+            home_geoid: int
+            age: int = feature(UserInfoDataset.age, default=0)
+            age_squared: str = feature(F("age") * F("age"))
+            credit_score: int
+
+    assert (
+        str(e.value)
+        == "Expected expression Ref('age') * Ref('age') for feature `age_squared` to return dtype `str` but got `int`"
+    )
```

Comment:
This shouldn't say `Ref` but `col`, same as Python syntax. I guess it's a Rust side change? If so, I can fix it as part of my PR.